cpp/src/spatial/linestring_distance.cu

```diff
@@ -101,19 +101,22 @@ double __device__ segment_distance_no_intersect_or_collinear(vec_2d<T> const& A,
  */
 template <typename T>
 double __device__
-segment_distance(vec_2d<T> const& A, vec_2d<T> const& B, vec_2d<T> const& C, vec_2d<T> const& D)
+segment_distance(vec_2d<T> const& a, vec_2d<T> const& b, vec_2d<T> const& c, vec_2d<T> const& d)
 {
-  double r_denom = (B.x - A.x) * (D.y - C.y) - (B.y - A.y) * (D.x - C.x);
-  double r_numer = (A.y - C.y) * (D.x - C.x) - (A.x - C.x) * (D.y - C.y);
-  if (r_denom == 0) {
+  auto ab    = b - a;
+  auto ac    = c - a;
+  auto cd    = d - c;
+  auto denom = det(ab, cd);
+
+  if (denom == 0) {
     // Segments parallel or collinear
-    return segment_distance_no_intersect_or_collinear(A, B, C, D);
+    return segment_distance_no_intersect_or_collinear(a, b, c, d);
   }
-  double r = r_numer / r_denom;
-  double s = ((A.y - C.y) * (B.x - A.x) - (A.x - C.x) * (B.y - A.y)) /
-             ((B.x - A.x) * (D.y - C.y) - (B.y - A.y) * (D.x - C.x));
+  double r_numer = det(ac, cd);
+  double r       = r_numer / denom;
+  double s       = det(ac, ab) / denom;
```

Comment:
Use auto, not double...

Reply:
Somehow when I look at the branch this line is already updated with `auto`.

Reply:
I probably clicked on a specific commit. Sorry.